audio_service/lib/audio_service.dart

```diff
@@ -1077,6 +1077,13 @@ class AudioService {
     );
   }
 
+  /// Update the AudioService configuration. Must be called after init.
+  /// Primarily intended to update how notification behaves during playback.
+  static Future<void> updateConfig(AudioServiceConfig config) async {
+    _config = config;
+    await _platform.configure(ConfigureRequest(config: config._toMessage()));
```

Comment:
Although this seems like a straightforward solution to the problem, I do want to take the time to think through how much of this config object should be moved to other state objects rather than be in a config passed to init.

Reply:
> 

Completely agree, I definitely took the path of least resistance here. I'd be happy to refactor the PR, to only update what we agree to be the dynamic options. However I also think clients will have a feel for what makes sense to update post-init.

As far as I can tell iOS only uses the `fastForwardInterval` and `rewindInterval` from the config, which both should be able to update dynamically.
For Android I don't see many that you should really need to change dynamically, maybe `androidNotificationIcon` and `androidShowNotificationBadge` could be relevant, but it becomes a little more subjective.

Reply:
On the Android side, the notification channel ID is one that requires some careful thought because in some advanced use cases are app may want to migrate from are old channel ID and name to a new one, which in turn involves deleting the old one. Maybe that is justification to move it into a separate API, or maybe it's not.

Reply:
Actually that renaming could be implemented within the same API since the old id can be found in the persistent config file which exists in the Android implementation.

Reply:
If I make sure to handle all property changes on Android, like the channel name, could you agree to keeping the API as proposed instead of making 8 different new setter methods?

Reply:
I wouldn't say it has to be one of those two extremer, when the ideal is probably somewhere in the middle. For example, I think `fastForwardInterval` and `rewindInterval` (and arguably also options like `androidNotificationOngoing`) could perhaps be moved into `PlaybackState` which is an existing class, not 3 new setter methods, and `androidStopForegroundOnPause` has a whole other discussion happening about how best to handle the Android service lifecycle. So in this manner, each one of those options in the config class could have a unique solution and at some point we need to go through each one carefully and consider how best to tackle it.

I think your PR might be a good short-term solution worth considering, though, before implementing a more radical API change later.